content/blog/2021-03-26-cdn-demo.md

```diff
@@ -0,0 +1,254 @@
+- **Users.** Any device that wants to connect (physical sensors, IoT devices, mobile devices, etc.)
+
+**Benefits of using Zenoh**
+- **Architecture.** Support for both hierarchical and flat organization, just by handling the configuration specified for Zenoh.
+- **Synchronization.** Zenoh natively provides the required features to synchronize contents although is also a pub/sub protocol.
+
+
+## Users involved
+
+Before talking of how to build a CDN, it’s crucial to understand the roles for the different users involved. Users may generally be divided into three levels of domain:
+
+**Infrastructure manager**
+- Perform the Initial configuration on the nodes
+- Deploy of the Zenoh routers
+- Create the backend for each Zenoh router
+
+**Admin**
+- Analyze the workload/traffic in order to decide whether to enable new storages
+- Create new storages on edge nodes
+
+**User**
+- Read/write data from/to Zenoh
+- Each user is always connected to the same edge node notwithstanding the data location
+
+
+## Demo steps
+
+Initial scenario:
+- Both origin and edge servers do not contain any data
+- Two nodes that contains connected Zenoh routers (1 origin server, 1 edge server) with backend-filesystem enabled
+- Backend created and available on both nodes.
+
+Steps:
+1. Admin creates the storage to the origin server
+2. A client writes data to the origin server by using the previously created storage:
+    - It creates a folder «fs_storage» and a text file «content»
+    - Other clients can access the «content» file, connecting from their edge servers
+3. Monitoring and migration alerting
+4. Admin creates the storage to the edge server
+    - Clients connected to that edge server retrieve the data locally
+
+Final scenario:
+- Both origin and edge servers contain the files within the folder «fs_storage/…»
+
+
+## Initial configuration
+
+Zenoh and the related projects in its ecosystem are open-source projects available on github. For this demo we use the following projects
+
+{{<table "table table-striped table-bordered">}}
+| Projects                   |  Overview                                        |
+|:---------------------------|:-------------------------------------------------|
+| **Zenoh**                  | Zero overhead pub/sub, store/query and compute protocol [[source code]](github.com/eclipse-zenoh/zenoh)  |
+| **Zenoh-backend-fielsystem** | Backend and Storages for zenoh using the file system [[source code]](github.com/eclipse-zenoh/zenoh-backend-filesystem) |
+| **Zenoh-python**            | Python client API for zenoh [[source code]](github.com/eclipse-zenoh/zenoh-python) |
+{{</table>}}
+
+
+and we initialize each node with the minimal set of tools to be used:
+
+1.	Install Cargo and Rust (nightly-build)
+2.	Build the “Zenoh” project and the “Zenoh-backend-filesystem” project
+    > **Warning:** the 2 project must be built with the same Rustc version to enable compatilities between them.
```

Comment:
Now all the zenoh projects contain a `rust-toolchain` file specifying the same Rustc version. It will be automatically be used when running `cargo build`.
=> Warning not needed